wordpress-mobile/WordPress-Android
# [Voice to Content] Implement frag/view model skeleton

Closes https://github.com/Automattic/wordpress-mobile/issues/75
Closes https://github.com/Automattic/wordpress-mobile/issues/74

This PR creates the skeleton classes for the Voice to Content feature. The intent is to have an entry point into the flow and be able to test FluxC calls.

Includes
- Add "Post to Audio" option to the My Site FAB (guarded by FF)
- Add the feature utils class to determine if the feature should be added to the FAB 
- Add a few resources
- Add a temporary audio file into the assets directory (this should be removed before release)
- Add the skeletons for VoiceToContentDialogFragment, VoiceToContentViewModel, and VoiceToContentUseCase

Note: No attempt was made to create production ready composables

-----
FAB Option | Entry view | Result view
-- | -- | --
<img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/f1598a81-3981-4437-ac3c-708491224c13"> | <img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/8ff6d218-ad44-43a5-b95d-59414e9ebd52">  | <img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/babba034-9a07-4c27-a547-221fbe323714">

## To Test:
**Prerequisites**
- Install and launch the app
- Login into a site in which you have access to create posts

**Test Post from audio not shown**
- Navigate to My Site
- Tap the FAB
- ✅ Verify the options do not include Post from Audio

**Test Post from audio is shown**
- Navigate to Me > Debug Settings
- Enable `voice_to_content` and restart the app
- Navigate to My Site
- Tap the FAB
- ✅ Verify the options do include Post from Audio

**Test Post will show the transcribed text**
- Run Test Post from audio is shown
- Tap the microphone
- ✅ Verify the microphone disappears and the text "Testing Jetpack AI. Testing" is shown in the bottom sheet
-----

## Regression Notes
1. Potential unintended areas of impact
The feature shows when the flag is off

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit and manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

-----

## PR Submission Checklist:

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones): N/A
